### ISSUE TITLE
useTransaction(...) not rolled back from thrown exceptions after version 3.6.0?

### ISSUE THREAD

**Issue by deneb-esper** [jdbi/jdbi#1960]
**Summary**: automatic rollbacks of transactions when an exception is encountered in `useTransaction(...)` work in 3.6.0, but don't work in any version above that.

I have a unit test to test rollbacks, that goes something like this:

```
assertThrows(Exception.class, () -> entityStore.insert(entity)); // make an exception is being thrown
assertTrue(entityStore.get(entityUuid) == null); // make sure the insert was rolled back
```

To keep things simple, I made entityStore's `insert` method manually throw an exception, like so:

```
jdbi.useTransaction(handle -> {
  handle.attach(entityDao.class).insert(entity);
  throw new Exception("Testing rollbacks");
});
```

Per my understanding of the documentation, if an exception is thrown from the `useTransaction` callback, then the transaction (anything operations that used the handle) should be rolled back.

The unit test passes when on version 3.6.0 (jdbi-bom), but if I upgrade to any version above that (I've tried 3.7.0 all the way to latest) the unit test fails on the `assertTrue`. Meaning the insert isn't being rolled back.

Was there a change in the way transaction rollbacks work after 3.6.0? I couldn't find anything immediately obvious in the release notes.

**Comment by stevenschlansker** [comment]
Hi @deneb-esper , I added a test case very similar to yours, and it passes:
https://github.com/jdbi/jdbi/pull/1966
So I think there's something more going on, unless I made a mistake transcribing your test. Is it possible that there's a driver bug, or some other change interfering with your rollbacks? Is it possible to reproduce the problem in a self-contained test case that we could look at?
